packages/react/src/TreeView/TreeView.module.css

```diff
@@ -0,0 +1,272 @@
+.TreeViewRootUlStyles {
+  padding: 0;
+  margin: 0;
+  list-style: none;
+
+  /*
+   * WARNING: This is a performance optimization.
+   *
+   * We define styles for the tree items at the root level of the tree
+   * to avoid recomputing the styles for each item when the tree updates.
+   * We're sacraficing maintainability for performance because TreeView
+   * needs to be performant enough to handle large trees (thousands of items).
+   *
+   * This is intended to be a temporary solution until we can improve the
+   * performance of our styling patterns.
+   *
+   * Do NOT copy this pattern without understanding the tradeoffs.
+   * Do NOT reference PRIVATE_* classnames outside of this file.
+   */
+  .PRIVATE_TreeViewItem {
```

Comment:
Don't know if we need these `PRIVATE_` names anymore since the CSS module hashes it?

Reply:
Good question, I was pretty iffy on this. You're right that there's no risk of these being copied since they're hashed now. Think keeping it would be more confusing than helpful at this point, so I'll go ahead and remove 👍

Reply:
Updated in https://github.com/primer/react/pull/5267/commits/0718102c37b6ada9e85eb4de2dced310f2c62985